hack/create-identity-secret.sh

```diff
@@ -26,5 +26,10 @@ cd "${REPO_ROOT}" && make "${KUBECTL##*/}"
 # shellcheck source=hack/parse-prow-creds.sh
 source "${REPO_ROOT}/hack/parse-prow-creds.sh"
 
+if "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null; then
```

Comment:
rather than exiting 0 early here (which might not be what we want if this script evolves in the future and more post-secret foo is added to the end of the flow) let's add a simply "if not exist" condition, like:

```sh
if ! "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null; then
  "${KUBECTL}" create secret generic "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" --from-literal=clientSecret="${AZURE_CLIENT_SECRET}"
fi
```

And then we can preserve the existing `kubectl label secret` command at the end, which is idempotent as currently implemented (`--overwrite=true`).

Thanks for making this improvement!

Reply:
Makes sense!